tests/test_bookmarks.py

```diff
@@ -0,0 +1,150 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+"""Tests for Bookmarks."""
+
+import pytest
+from selenium.webdriver.remote.webdriver import WebDriver
+from foxpuppet.windows import BrowserWindow
+from foxpuppet.windows.browser.bookmarks.bookmark import (
+    Bookmark,
+    BookmarkData,
+)
+
+
+class BookmarkEntry:
+    """Class representing a Firefox bookmark entry."""
+
+    def __init__(self, url: str, name: str = "", keyword: str = "", tags: list[str] = []):
+        self.url = url
+        self.name = name
+        self._keyword = keyword
+        self._tags = tags or []
+
+    @property
+    def keyword(self):
+        """Returns the current list of keywords."""
+        return self._keyword
+
+    @keyword.setter
+    def keyword(self, new_keyword: str):
+        """Sets new keywords for the bookmark."""
+        if isinstance(new_keyword, str):
+            self._keyword = new_keyword
+        else:
+            raise ValueError("Keyword must be a string.")
+
+    @property
+    def tags(self):
+        """Returns the current list of tags."""
+        return self._tags
+
+    @tags.setter
+    def tags(self, new_tags: list[str]):
+        """Sets new tags for the bookmark."""
+        if isinstance(new_tags, list) and all(isinstance(t, str) for t in new_tags):
+            self._tags = new_tags
+        else:
+            raise ValueError("Tags must be a list of strings.")
```

Comment:
This seems like the wrong place to do this validation, shouldn't it be within the `tags` property? Also, shouldn't our static typing alert to an improperly set tag?